README.md
# tutorialtime README

This is the README for your extension "tutorialtime". After writing up a brief description, we recommend including the following sections.

## Features

Describe specific features of your extension including screenshots of your extension in action. Image paths are relative to this README file.

For example if there is an image subfolder under your extension project workspace:

\!\[feature X\]\(images/feature-x.png\)

> Tip: Many popular extensions utilize animations. This is an excellent way to show off your extension! We recommend short, focused animations that are easy to follow.

## Requirements

If you have any requirements or dependencies, add a section describing those and how to install and configure them.

## Extension Settings

Include if your extension adds any VS Code settings through the `contributes.configuration` extension point.

For example:

This extension contributes the following settings:

* `myExtension.enable`: enable/disable this extension
* `myExtension.thing`: set to `blah` to do something

## Known Issues

Calling out known issues can help limit users opening duplicate issues against your extension.

## Release Notes

Users appreciate release notes as you update your extension.

### 1.0.0

Initial release of ...

### 1.0.1

Fixed issue #.

### 1.1.0

Added features X, Y, and Z.

-----------------------------------------------------------------------------------------------------------

## Working with Markdown

**Note:** You can author your README using Visual Studio Code.  Here are some useful editor keyboard shortcuts:

* Split the editor (`Cmd+\` on macOS or `Ctrl+\` on Windows and Linux)
* Toggle preview (`Shift+CMD+V` on macOS or `Shift+Ctrl+V` on Windows and Linux)
* Press `Ctrl+Space` (Windows, Linux) or `Cmd+Space` (macOS) to see a list of Markdown snippets

### For more information

* [Visual Studio Code's Markdown Support](http://code.visualstudio.com/docs/languages/markdown)
* [Markdown Syntax Reference](https://help.github.com/articles/markdown-basics/)

**Enjoy!**


TutorialTime (FAIL)/CHANGELOG.md
# Change Log

All notable changes to the "tutorialtime" extension will be documented in this file.

Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.

## [Unreleased]

- Initial release

tictactoe.py
class Grid:
  def __init__(self, length):
    self.grid = [[[0,0] for x in range(0,length)] for y in range(0,length)]

  def __len__(self):
    return len(self.grid)

  def __getitem__(self, key):
    return self.grid[key]

  def __str__(self):
    output, length = "", len(self.grid)
    for i in range(0, length):
      for j in range(0, length):
        if self.grid[i][j][0] == 1:
          output += "X"
        elif self.grid[i][j][1] == 1:
          output += "O"
        else:
          output += " "
        if j < length - 1:
          output += "|"
      if i < length - 1:
        output += "\n" + "--" * length  + "\n"
    return output

  def setGridBlock(self, x, y, value):
    if self.grid[x][y][1-value] == 0 and self.grid[x][y][value] == 0:
      self.grid[x][y][value] = 1
      self.grid[x][y][1-value] = 0
      return True
    return False

  def isWin(self, val):
    length = len(self.grid)
    for i in range(0,length):
      col = [self.grid[x][i] for x in range(0, length)]
      if self.checkRowCol(self.grid[i], val) or self.checkRowCol(col, val):
        return True
    firstDiagonal = [self.grid[x][x] for x in range(0, length)]
    secondDiagonal = [self.grid[length-1-x][x] for x in range(0,length)]
    if self.checkRowCol(firstDiagonal, val) or self.checkRowCol(secondDiagonal, val):
      return True
    return False

  def isStalemate(self):
    length = len(self.grid)
    firstDiagonal = [self.grid[x][x] for x in range(0, length)]
    secondDiagonal = [self.grid[length-1-x][x] for x in range(0,length)]
    for i in range(0,length):
      col = [self.grid[x][i] for x in range(0, length)]
      if self.checkStale(self.grid[i]) and self.checkStale(col) and self.checkStale(firstDiagonal) and self.checkStale(secondDiagonal):
        return True
    return False
  
  def checkStale(self, arr):
    numX, numO = 0, 0
    for i in range(0, len(arr)):
      if arr[i][0] == 1:
        numX += 1
      if arr[i][1] == 1:
        numO += 1
      if numX > 0 and numO > 0:
        return True
    return False
      
  def checkRowCol(self, arr, val):
    for i in range(0, len(arr)):
      if arr[i][val] == 0:
        return False
    return True


def playGame(length):
  board = Grid(length)
  player = 0
  while not board.isWin(0) and not board.isWin(1):
    print(board)
    print("Player " + str(player + 1) + " turn")
    changeCoordinates(board, player)
    if board.isWin(player):
      print("=====================================")
      print("Player" + str(player + 1) + " wins!")
      print("=====================================")
      print(board)
      return
    if board.isStalemate():
      print("=====================================")
      print("Stalemate!")
      print("=====================================")
      print(board)
      return
    player = 1 - player

def changeCoordinates(board, player):
  x,y = askCoordinates(board)
  while not board.setGridBlock(int(x),int(y),player):
    print("This move is not possible!")
    x,y = askCoordinates(board)
    board.setGridBlock(int(x),int(y),player)

def askCoordinates(board):
  x = int(input("Choose a x coordinate: "))
  while x < 0 or x >= len(board):
    print("Coordinate does not fit specified range of 0 to " + str(len(board)-1))
    x = int(input("Choose a x coordinate: "))
  y = int(input("Choose a y coordinate: "))
  while y < 0 or y >= len(board) :
    print("Coordinate does not fit specified range of 0 to " + str(len(board)-1))
    y = int(input("Choose a y coordinate: "))
  return x,y


playGame(3)


TutorialTime (FAIL)/src/extension.ts

import * as vscode from 'vscode';

// your extension is activated the very first time the command is executed
export function activate(context: vscode.ExtensionContext) {
	context.subscriptions.push(
		vscode.commands.registerCommand('tutorialtime.start', () => {
			const panel = vscode.window.createWebviewPanel('tutorialtime', 'Tutorial', vscode.ViewColumn.One, {enableScripts: true});
			panel.webview.html = getWebviewContent();
		})
	);

	

}

function getWebviewContent() {
	return `<!DOCTYPE html>
			<html lang="en">
				<head>
					<meta charset="UTF-8">
					<meta name="viewport" content="width=device-width, initial-scale=1.0">
					<title>Cat Coding</title>
				</head>
				<body
				<h1>Hello</h1>
				<iframe width="1280" height="720" 
						src="https://www.youtube.com/embed/dQw4w9WgXcQ?rel=0&amp;showinfo=0"
						frameborder="0" allowfullscreen>
				</iframe>
					
				</body>
			</html>`;
}



// this method is called when your extension is deactivated
export function deactivate() {}


TutorialTime (FAIL)/out/extension.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deactivate = exports.activate = void 0;
const vscode = require("vscode");
// your extension is activated the very first time the command is executed
function activate(context) {
    context.subscriptions.push(vscode.commands.registerCommand('tutorialtime.start', () => {
        const panel = vscode.window.createWebviewPanel('tutorialtime', 'Tutorial', vscode.ViewColumn.One, { enableScripts: true });
        panel.webview.html = getWebviewContent();
    }));
}
exports.activate = activate;
function getWebviewContent() {
    return `<!DOCTYPE html>
			<html lang="en">
				<head>
					<meta charset="UTF-8">
					<meta name="viewport" content="width=device-width, initial-scale=1.0">
					<title>Cat Coding</title>
				</head>
				<body
				<h1>Hello</h1>
				<iframe width="1280" height="720" 
						src="https://www.youtube.com/embed/dQw4w9WgXcQ?rel=0&amp;showinfo=0"
						frameborder="0" allowfullscreen>
				</iframe>
					
				</body>
			</html>`;
}
// this method is called when your extension is deactivated
function deactivate() { }
exports.deactivate = deactivate;
//# sourceMappingURL=extension.js.map